lib/assert.js

```diff
@@ -163,21 +173,26 @@ x._snapshot = function (tree, optionalMessage, match, snapshotStateGetter) {
 		snapshotState: state
 	};
 
-	const result = toMatchSnapshot.call(context, tree);
+	const result = toMatchSnapshot.call(context, JSON.stringify(tree));
 
-	let message = 'Please check your code or --update-snapshots\n\n';
+	let message = 'Please check your code or --update-snapshots';
 
 	if (optionalMessage) {
-		message += indentString(optionalMessage, 2);
-	}
-
-	if (typeof result.message === 'function') {
-		message += indentString(result.message(), 2);
+		message += '\n\n' + indentString(optionalMessage, 2);
 	}
 
 	state.save();
 
-	test(result.pass, create(result, false, 'snapshot', message, x.snap));
+	let expected;
+
+	if (result.expected) {
+		// JSON in a snapshot is surrounded with `"`
```

Comment:
@vadimdemedes Can you include the link in the comment. Usually comments are great for explaining "why".